src/coreclr/src/jit/codegen.h

```diff
@@ -1082,7 +1082,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         bool TestImmOpZeroOrOne() const
```

Comment:
Is the intent of this to cover `0` or `1` or just to cover an `if/else` scenario?

Reply:
TestImmOpZeroOrOne() is true when imm can only be 0 or 1, i.e. when we don't need to explicitly compute target branch value (and trash a register to hold this value).

Reply:
Right, but I was trying to understand if that would also be the case if the value could only be `1` or `2` (for example).

Reply:
No - the codegen uses cbnz to test whether immOpReg is non zero. In your case, when both values are be non-zero such approach would not work.